system/HTTP/IncomingRequest.php

```diff
@@ -490,6 +490,20 @@ public function getOldInput(string $key)
 		{
 			return $_SESSION['_ci_old_input']['get'][$key];
 		}
+
+		helper('array');
+
+		// Check for an array value in POST.
+		if (isset($_SESSION['_ci_old_input']['post']) && dot_array_search($key, $_SESSION['_ci_old_input']['post']))
```

Comment:
You're doing the dot array search twice here which is inefficient. More inefficient than assigning it to a variable, I believe. Please change that.